OsmAnd-java/src/main/java/net/osmand/util/Algorithms.java

```diff
@@ -1372,6 +1393,25 @@ public static <T> List<T> asOneList(Collection<T> ... collections) {
 		return result;
 	}
 
+	public static <T> T searchElementWithCondition(Collection<T> collection, CallbackWithObject<T> condition) {
```

Comment:
Are these methods needed? We don't use them, and we have similar methods in Kotlin